### DISCUSSION TITLE
Sometimes not work for vim-tmux-navigator keybinding.

### DISCUSSION THREAD

**Discussion by OriLiMu** [LazyVim/LazyVim#4109] [Q&A]
# update
when I directly run the command `TmuxNavigateLeft` it works fine. I add the key map to the `keymap.lua` file. I can check the map by `map <a-l>` seems fine but still not work.
I disable all the plugins and remove all customize keymap not work. 
I disable my `init.lua` file not works. 

When I press `alt - l ` I get this on the status bar. But I don't know where it is defined. I kill my keymap software in system wide not works. 
![image](https://github.com/user-attachments/assets/87674e4e-f9cc-44d2-a5a3-6e59dde95bab)

Here is the map of <a-l>
![image](https://github.com/user-attachments/assets/7f1ac3bb-f172-43ca-a7d9-498d1a1e3b6b)

Where is the keymap for `alt-l` ? I cant find it. I use `ag <A-l>` or `ag <M-l>` search in directory `~/.local/share/nvim/lazy/lazy.nvim` not found related. 

---
I have vim-tmux-navigator setting like this:
```lua
  {
    "christoomey/vim-tmux-navigator",
    lazy = false,
    cmd = {
      "TmuxNavigateLeft",
      "TmuxNavigateDown",
      "TmuxNavigateUp",
      "TmuxNavigateRight",
      "TmuxNavigatePrevious",
    },
    keys = {
      { "<a-h>", "<cmd><C-U>TmuxNavigateLeft<cr>" },
      { "<c-j>", "<cmd><C-U>TmuxNavigateDown<cr>" },
      { "<c-k>", "<cmd><C-U>TmuxNavigateUp<cr>" },
      { "<a-l>", "<cmd><C-U>TmuxNavigateRight<cr>" },
      { "<c-\\>", "<cmd><C-U>TmuxNavigatePrevious<cr>" },
    },
  },
```

which use a-l, a-h, to jump between tmux pane and nvim. But when I enter the nvim. Press a-l, a-l usually not work. 

On the status bar it shows. 
![image](https://github.com/user-attachments/assets/7f1e0473-e96d-42dd-9603-85458bc00545)

I try the command `map <a-l>` saids no mapping. 

Sometimes I run the command `:TmuxNavigateLeft` then jump back it will work just for sometime. So how to solve my issue now . 

The c-j, c-k to move up / down works fine.

**Comment by zaynOm** [comment]
I'm facing the same issue, it seams like Lazyvim has already some keybindings to navigate between windows inside nvim.

let me know if you found any solution

**Reply by zaynOm** [reply]
just found a solution, you can simply add the keymaps in `nvim/lua/config/keymaps.lua`

```lua
vim.keymap.set("n", "<C-h>", "<Cmd>NvimTmuxNavigateLeft<CR>", {})
vim.keymap.set("n", "<C-j>", "<Cmd>NvimTmuxNavigateDown<CR>", {})
vim.keymap.set("n", "<C-k>", "<Cmd>NvimTmuxNavigateUp<CR>", {})
vim.keymap.set("n", "<C-l>", "<Cmd>NvimTmuxNavigateRight<CR>", {})
```

**Reply by robbash** [reply]
Great solution, thanks!

I found that mentioned "sometimes" in my case was opening nvim with a filename given, while it works nicely when starting nvim without any files.

**Reply by Meursault667** [reply]
> just found a solution, you can simply add the keymaps in `nvim/lua/config/keymaps.lua`
> 
> ```lua
> vim.keymap.set("n", "<C-h>", "<Cmd>NvimTmuxNavigateLeft<CR>", {})
> vim.keymap.set("n", "<C-j>", "<Cmd>NvimTmuxNavigateDown<CR>", {})
> vim.keymap.set("n", "<C-k>", "<Cmd>NvimTmuxNavigateUp<CR>", {})
> vim.keymap.set("n", "<C-l>", "<Cmd>NvimTmuxNavigateRight<CR>", {})
> ```

after a whole afternoon of trying other ways, this is the first, that realy works great ( when you delete "Nvim" in "~~Nvim~~TmuxNavigateX")
